### DISCUSSION TITLE
Consider rendering dependency nodes with html text instead of drawing on canvas.

### DISCUSSION THREAD

**Discussion by JoeyCarson** [madushans/DependenSee#35] [Ideas]
When you have a complex dependency graph, the existing implementation makes things harder to search by Ctrl+F. When I inspect the nodes, they are drawn on canvas. Drawing on canvas provides a reasonable way to draw complex graphics, but it makes the DOM less searchable which can make insights harder to determine easily as the dependency graph becomes scarier.

I'm not front end dev myself, so considering the complexity of rendering a graph with html, it may be easier to simply add a search function which could illuminate just those package nodes that match the search text.

**Comment by madushans** [comment]
Yea a search function may be easier. I looked into making it DOM based, but there are a few issues there that I don't want to deal with.

- If you have enough nodes, (some have upwards of 100) the page can become real slow.
- The graphing library used (vis.js) doesn't seem to support DOM based graphs. So have to write it by hand.

Current plan is to beef up the HTML visualization in the [upcoming V3](https://github.com/madushans/DependenSee/blob/main/docs/Version3.md) and allow search and filtering in it that can hopefully highlight matching nodes given some text or regex. Though admittedly I haven't spent much time on it lately, so I don't have a target release date.
